### ISSUE TITLE
Display all values for metadata on work show page

### ISSUE THREAD

**Issue by rjkati** [samvera/hyrax#6276]
### Descriptive summary

On pg.nurax and koppie, several non-required metadata fields do not display on the work show page even though they contain values.

All metadata fields with values display on the work show page on dev.nurax

### Steps to reproduce the behavior in User Interface (UI)

1. On pg.nurax, go to Dashboard -> Works -> Add New Work -> Generic Work
2. Fill out all metadata fields with values, including the fields under the Additional Fields button
3. Upload a file, check the deposit agreement and click Save
4. Observe that not all the fields for which you entered values display on the work show page

### Actual behavior (include screenshots if available)

Only a handful of fields display on the work show page. Other fields which contain values do not display.

Example of a GenericWork which has all metadata fields filled out: https://pg.nurax.samvera.org/concern/generic_works/98f5577d-2eb9-420e-bfeb-f6c38f3e785a?locale=en

For GenericWorks, the following fields do **not** display on pg.nurax but do display on dev.nurax

- abstract
- access rights
- alternative title
- location
- contributor
- identifier
- publisher
- label
- language
- license
- rights notes
- source

### Acceptance Criteria/Expected Behavior

All metadata fields which contain values should display on the work show page

### Related work
#5887 
#5015

**Comment by hackartisan** [comment]
label does not appear in the form on dev.nurax and doesn't appear to display when I index it; I'm not going to include it.

**Comment by hackartisan** [comment]
Location (based near) is blocked by https://github.com/samvera/hyrax/issues/6277.

**Comment by jlhardes** [comment]
The Location field is still a problem on pg.nurax. I created a new work and added metadata to all fields but the Location field didn't offer me a drop-down (see screenshot).

![Screenshot 2023-10-02 at 10 13 12 AM](https://github.com/samvera/hyrax/assets/3440166/1e13e6eb-d2e7-41a8-a028-b0c249a8c525)

I saved the work without Location and then went back in to Edit. There still wasn't a field available for Location so I clicked the link to "+ Add another Location" and a drop-down field appeared. I added a location and saved but the Location field didn't show in the Work view page (all other fields are showing). I tried to Edit again and received the following error (see screenshot): NoMethodError in Hyrax::GenericWorks#edit

![Screenshot 2023-10-02 at 10 16 21 AM](https://github.com/samvera/hyrax/assets/3440166/f913ff6e-cd51-45b2-a5fe-a9abbeff9dc2)

This is the work on pg.nurax: https://pg.nurax.samvera.org/concern/generic_works/c9915667-040f-4761-b32e-70ef82fe1488?locale=en
